nemo-physical/src/lib.rs

```diff
@@ -18,6 +18,7 @@
     clippy::cast_possible_truncation,
     clippy::cast_possible_wrap
 )]
+#![allow(clippy::single_range_in_vec_init)]
```

Comment:
Please only allow this on the lines where this currently happens.

Reply:
Well as discussed above, I do not find this lint particularly useful as it is meant more as a UX improvement, to help if beginners that do not understand the error message of the type system. It does not really catch any new errors.

Additionally we happen to use `Vec<Range>` pretty often, so the code could become cluttered.
I'd say it's safe to disable this lint.

Reply:
Basically what I'm saying is this Lint will only ever give us false-positives.

Reply:
Counterpoint: we have a lot of `Vec<Range<…>>` in our code, yet we only have three occurrences of this pattern, all of them in tests, so this clearly doesn't lead to _many_ false-positives. I also don't see why this wouldn't be a relevant warning for non-`Range`-typed sequences.

To avoid clutter, though, instead of allowing this lint, just be explicit about creating single-element Vecs:
`vec![3..9; 1]` does _not_ trigger the lint _and_ communicates intent.

Reply:
Agreed, `vec![3..9; 1]` is a valid solution I did not think of (although I'd be worried, that cargo fmt someday decides to "clean up" the unnecessary length).

I also don't find the lint entirely useless, but it seems to strong currently. I think I'll file an issue with clippy and in the mean time go with the solution you proposed.